spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -81,6 +83,10 @@
 
 	private volatile AbstractMessageChannelMetrics channelMetrics = new DefaultMessageChannelMetrics();
 
+	public AbstractMessageChannel() {
+		this.interceptors = new ChannelInterceptorList(logger);
+	}
```

Comment:
Right, I've been reading too much about compilers re-ordering stuff lately :smile: so I felt better to be safe.